eng/tools/lint-diff/src/markdown-utils.ts

```diff
@@ -17,91 +11,32 @@ export enum MarkdownType {
 
 /**
  *
- * @param markdownFile Path to the markdown file to parse
+ * @param readme Readme object to extract
  * @returns {Promise<MarkdownType>} The type of OpenAPI spec in the markdown file, or "default" if not found
  */
-export async function getOpenapiType(markdownFile: string): Promise<MarkdownType> {
-  let markdownContent = await readFile(markdownFile, { encoding: "utf-8" });
-  for (const codeBlock of parseCodeblocks(markdownContent)) {
-    if (
-      !codeBlock.info ||
-      codeBlock.info.trim().toLocaleLowerCase() !== "yaml" ||
-      !codeBlock.literal
-    ) {
-      continue;
-    }
-
-    let lines = codeBlock.literal.trim().split("\n");
-
-    for (const line of lines) {
-      if (line.trim().startsWith("openapi-type:")) {
-        let openapiType = line.trim().split(":")[1].trim().toLowerCase();
-
-        if (Object.values(MarkdownType).includes(openapiType as MarkdownType)) {
-          return openapiType as MarkdownType;
-        }
-      }
-    }
+// TODO: Should this be placed in the Readme class?
+export async function getOpenapiType(readme: Readme): Promise<MarkdownType> {
```

Comment:
This heuristic is used in Legacy LintDiff but it could easily go in `readme.js` 

Reasons to put in readme.js: 
* Reusable 

Reasons against putting in readme.js: 
* It'd be a widely shared heuristic, changes to the heuristic could break dependant apps

Reply:
Although, what's the value of converting this from string to enum?

Reply:
The type does eventually get turned into a string when used. There's no real value other than constraining the possible output values. Doing that with strings would certainly make the code more readable.

Reply:
[#34836](https://github.com/Azure/azure-rest-api-specs/issues/34836)